lib/compilers/glimmer/index.js

```diff
@@ -13,14 +13,39 @@ export default class GlimmerTemplatePrecompiler extends Filter {
 
   processString(content, relativePath) {
     let specifier = getTemplateSpecifier(this.rootName, relativePath);
-    return `export default ${this.precompile(content, { meta: { specifier, '<template-meta>': true } })};`;
+    let { contentWithReplacedPragmaTag, managerId } = extractPragmaTag(content);
+    let meta = { specifier, '<template-meta>': true };
+
+    if (managerId) {
+      meta.managerId = managerId;
+    }
+
+    let precompiledContent = this.precompile(contentWithReplacedPragmaTag, { meta });
+
+    return `export default ${precompiledContent};`;
   }
 
   precompile(content, options) {
     return precompile(content, options);
```

Comment:
inside of [`precompile` call](https://github.com/glimmerjs/glimmer-vm/blob/9fb4cd315589186ce7179bed4015d56764001ee7/packages/%40glimmer/compiler/lib/compiler.ts#L56), we call [`preprocess`](https://github.com/glimmerjs/glimmer-vm/blob/9fb4cd315589186ce7179bed4015d56764001ee7/packages/%40glimmer/syntax/lib/parser/tokenizer-event-handlers.ts#L298). Then the plugin [`transform method`](https://github.com/glimmerjs/glimmer-vm/blob/9fb4cd315589186ce7179bed4015d56764001ee7/packages/%40glimmer/syntax/lib/parser/tokenizer-event-handlers.ts#L308) gets executed. Note, this all happens before calling [`Template.compile`](https://github.com/glimmerjs/glimmer-vm/blob/9fb4cd315589186ce7179bed4015d56764001ee7/packages/%40glimmer/compiler/lib/compiler.ts#L62) and we can mutate options inside of `transform`.